x-pack/plugins/code/server/indexer/lsp_incremental_indexer.ts

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import fs from 'fs';
+import util from 'util';
+
+import { ProgressReporter } from '.';
+import { Diff, DiffKind } from '../../common/git_diff';
+import { toCanonicalUrl } from '../../common/uri_util';
+import {
+  Document,
+  IndexStats,
+  IndexStatsKey,
+  LspIncIndexRequest,
+  RepositoryUri,
+} from '../../model';
+import { GitOperations } from '../git_operations';
+import { EsClient } from '../lib/esqueue';
+import { Logger } from '../log';
+import { LspService } from '../lsp/lsp_service';
+import { ServerOptions } from '../server_options';
+import { detectLanguage } from '../utils/detect_language';
+import { LspIndexer } from './lsp_indexer';
+import { DocumentIndexName, ReferenceIndexName, SymbolIndexName } from './schema';
+
+export class LspIncrementalIndexer extends LspIndexer {
```

Comment:
so add a todo to delete LspIndexer, or get rid of the processRequest and other logic in LspIndexer

Reply:
LspIndexer is still applied for the very initial index request.